lib/cartopy/io/shapereader.py

```diff
@@ -253,10 +258,23 @@ def records(self):
                                item.items() if key != 'geometry'})
 
 
-if _HAS_FIONA:
-    Reader = FionaReader
-else:
-    Reader = BasicReader
+def Reader(filename, bbox=None, **kwargs):
```

Comment:
Is there a reason to make this change as well? Now `Reader` is a function rather than a class.

Reply:
I don't think Reader being an alias for BasicReader / FionaReader is ideal. Using a function avoids this and also adds the docstring.

Reply:
> I don't think Reader being an alias for BasicReader / FionaReader is ideal.

I agree it isn't ideal. But, I personally would prefer that to a function that looks like a class, I won't block on it, but I also won't approve it either.

Another way of doing this would have been to create a `class Reader` that stores an instance of `self._reader = FionaReader() if _HAS_FIONA else BasicReader()` and then dispatches out the methods to whatever `self._reader` it contains, but that seems like more work than is necessary as well unless we start getting more readers.

Also, I think you can add a docstring right below the `Reader` definition like we do for the configuration dictionary: https://github.com/SciTools/cartopy/blob/96ee0a3b4ab134b58e58020620f88624c3debf3a/lib/cartopy/__init__.py#L23-L31
but maybe to get that to work you'd need to put it all on one line?
```python
Reader = FionaReader if _HAS_FIONA else BasicReader
"""Explanation of Reader"""
```

Reply:
Thanks for the tip on adding a docstring to the existing Reader line! I like that solution, which has now been made.